File: Interviews/LeetCode/Daily-Problems/2022/October/Day-28/49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> ump;
        for (string s : strs) {
            string t = s;
            sort(t.begin(), t.end());
            ump[t].push_back(s);
        }

        vector<vector<string>> ans(ump.size());
        int j = 0;
        for (auto i : ump) {
            for (string s : i.second) {
                ans[j].push_back(s);
            }
            j += 1;
        }

        return ans;
    }
};

File: Interviews/LeetCode/Topic-Wise/Strings/Medium/Group-Anagrams/49.py
class Solution:
    def groupAnagrams (self, strs: List[str]) -> List[List[str]]:
        ump, ans = {}, []
        for s in strs:
            t = ''.join(sorted(s))
            if t not in ump:
                temp = []
                temp.append(s)
                ump[t] = temp
            else:
                temp = ump.get(t)
                temp.append(s)
                ump[t] = temp
        for value in ump.values():
            temp = []
            for s in value:
                temp.append(s)
            ans.append(temp)
        
        return ans